packages/bvaughn-architecture-demo/src/contexts/SelectedFrameContext.tsx

```diff
@@ -0,0 +1,28 @@
+import { FrameId, PauseId } from "@replayio/protocol";
+import { createContext, PropsWithChildren, useMemo } from "react";
+import useCurrentPause from "../hooks/useCurrentPause";
+
+export interface SelectedFrameContextType {
+  selectedFrameId: {
+    pauseId: PauseId;
+    frameId: FrameId;
+  } | null;
+}
+
+export const SelectedFrameContext = createContext<SelectedFrameContextType>({
+  selectedFrameId: null,
+});
+
+export function SelectedFrameContextRoot({ children }: PropsWithChildren<{}>) {
+  const pause = useCurrentPause();
```

Comment:
`useCurrentPause` _suspends_, so this change means that we might Suspend pretty much at the top of the DevTools tree. I'm a little wary of that. If we mess up an update (don't wrap it in a transition) then it could cause the entire app to unmount and remount.

Maybe we should add a guard here?

What if we replaced `useCurrentPause` with a new hook, `usePauseData`, and then we passed in a deferred pause to that hook here to avoid accidentally suspending during a high priority update?

Reply:
Thanks, I overlooked that (and will fix it). I think we should add `Suspense` to the names of all functions that can suspend.

Reply:
Yeah, that's probably a good call. This hook should either be called `useCurrentPauseSuspends` to make it more obvious, or we should just remove the _suspending_ part.

Maybe we could split this hook into something smaller that _doesn't_ suspend but is still helpful, like...
```js
function useExecutionPointIsLoaded(executionPoint: ExecutionPoint): boolean {
  const client = useContext(ReplayClientContext);
  const loadedRegions = useLoadedRegions(client);

  const isLoaded = loadedRegions !== null && isPointInRegions(executionPoint, loadedRegions.loaded);
  return isLoaded;
}
```

Then you could decide whether to Suspend or not like
```js
const client = useContext(ReplayClientContext);
const { executionPoint } = useContext(TimelineContext);
const isLoaded = useExecutionPointIsLoaded(executionPoint);
let pause = null;
if (isLoaded) {
  pause = getPauseForExecutionPoint(client, executionPoint);
}
```

Reply:
But `getPauseForExecutionPoint()` can still suspend in that case because we may not have a `pauseId` for that execution point yet (and have to call `Session.createPause` to get it).

Reply:
What if the context offered a suspending function `getSelectedFrameId()`, so it wouldn't suspend itself, only the components that use the function would...?

Reply:
> But `getPauseForExecutionPoint()` can still suspend in that case because we may not have a `pauseId` for that execution point yet (and have to call `Session.createPause` to get it).

Yes. That's why I was suggesting that part _doesn't_ go into a hook. (So it's _clear_ when you choose to Suspend.)

Basically, I think having a hook that _suspends_ was probably not a good idea.

> What if the context offered a suspending function `getSelectedFrameId()`, so it wouldn't suspend itself, only the components that use the function would...?

I'm not sure what you're suggesting.

Reply:
> I'm not sure what you're suggesting.

What I did in this commit: https://github.com/replayio/devtools/pull/7987/commits/df483c26e594195efa3aa77c0fe10cac2e4a9b1c